code-journal/src/ViewEntries.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import { Entry, readEntries } from './data.ts';
+import { Link, useNavigate } from 'react-router-dom';
+import { FaPencilAlt } from 'react-icons/fa';
+
+export function ViewEntries() {
+  const [entries, setEntries] = useState<Entry[]>();
+  const navigate = useNavigate();
+  useEffect(() => {
+    async function fetchEntries() {
+      try {
+        const fetchedEntries = await readEntries();
```

Comment:
You should indicate to the user that loading is happening. Otherwise if there are network delays the user will conclude something is broken and will leave your site.